web/src/components/Education/Education.jsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react';
+import styles from '../Education/Education.module.css';
+
+export const Education = () => {
+  const [education, setEducation] = useState({
+    institution: '',
+    program: '',
+    startDate: '',
+    endDate: '',
+  });
+
+  const [error, setError] = useState({});
+
+  const validateEducation = () => {
+    const newErrors = {};
+    if (!education.institution.trim())
+      newErrors.institution = 'Institution is required';
+    if (!education.program.trim()) newErrors.program = 'Program is required';
+
+    if (!education.startDate) newErrors.startDate = 'Start date is required';
+    if (!education.endDate) newErrors.endDate = 'End date is required';
+
+    if (education.startDate && education.endDate) {
+      const start = new Date(education.startDate);
+      const end = new Date(education.endDate);
+
+      if (end <= start) {
+        newErrors.endDate = 'End date must be after start date';
+      }
+    }
+    return newErrors;
+  };
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+
+    setEducation((prev) => ({
+      ...prev,
+      [name]: value,
+    }));
+
+    const validateError = validateEducation();
+    setError(validateError);
```

Comment:
similar to @Priscilla-MacGatus [PR](https://github.com/SyedArslanHaider/smart-cv-builder/pull/45/files#diff-a8d3587f37d98968897932a42d809c01d96ed1e0ed7eda95b5687d0c69f7b181R22-R30) you should make some changes here so that we dont validate on every change which can be a bit much but instead on blur and reset the error when the user starts to edit again.

Have a look at `web/src/components/ProfessionalSummary/ProfessionalSummary.jsx` in that PR to see the changes that should happen